### PR TITLE
Trigger a new build API entry point

### PR DESCRIPTION
**Pull request by kkorekk**
Trigger new build using HTTP endpoint. For example:
```
http://katana:8001/json/builders/proj55-ParentA_ChildA_GrandChildA/start-build/ 
```
EXAMPLE POST REQUEST (for more options please look at JSON help page):
``` json
{
	"force_chain_rebuild": true,
	"force_rebuild": true,
	"owner": "Kornel Maleszka <kornel.maleszka@stxnext.pl>",
	"selected_slave": "allCompatible"
}
```
RESPONSE:
```json
[
    {
        "build_request_id": 9
    },
    {
        "build_request_id": 8
    }
]
```

TO DO:
- [x] - handle multiple slaves
- [x] - unit tests
- [x] - documentation
- [x] - handle custom properties in api endpoint

### CODE REVIEW

File: master/buildbot/status/web/status_json.py [hunk 1]
```diff
@@ -127,6 +129,44 @@
...
+    
+    - }
+
+    - This endpoint will return the one element list of build request objects,
+if 'allCompatible' was used then the list will contain build request object for all slaves.
+
+    - Example:
+    
+    - [{"build_request_id": 1}] 
+>>>>>>> 105619308... added documentation for endpoint
```
Comment (mariangemarcano) [inline master/buildbot/status/web/status_json.py]: this line seems odd, maybe a left over of a merging conflict ?

File: master/buildbot/status/web/status_json.py [hunk 2]
```diff
@@ -434,30 +474,132 @@ def asDict(self, request):
...
 
 
+class StartBuildJsonResource(AccessorMixin, resource.Resource):
+    help = 'Start a new build'
+    isLeaf = True
+    pageTitle = 'Start a new build'
+    schema = {
+        'type': 'object',
+        'properties': {
+            'force_chain_rebuild': {'type': 'boolean'},
```
Comment (mariangemarcano) [inline master/buildbot/status/web/status_json.py]: I guess if the dynamic properties from the force scheduler changed this could get out of sync.

**Review by mariangemarcano — APPROVED** [review]

File: master/buildbot/status/web/status_json.py (hunk 1 above)
Comment (kkorekk) [reply master/buildbot/status/web/status_json.py]: Yep, I haven't noticed this. :)

File: master/buildbot/status/web/status_json.py (hunk 2 above)
Comment (kkorekk) [reply master/buildbot/status/web/status_json.py]: These are the default properties, but this endpoint will not accept custom properties for builder, I'll investigate it.

File: master/buildbot/schedulers/manager.py [hunk 3]
```diff
@@ -19,13 +19,29 @@
...
+                return scheduler
+
+    def findSchedulerByBuilderName(self, builder_name, scheduler_type=None):
+        for scheduler in self.services:
+            if builder_name in scheduler.builderNames:
+                if scheduler_type is None:
+                    return scheduler
+                elif isinstance(scheduler, scheduler_type):
+                    return scheduler
+                else:
```
Comment (campos-ddc) [inline master/buildbot/schedulers/manager.py]: `else: continue` is doing nothing here, you can remove it if you want

**Review by campos-ddc — APPROVED** [review]

**Pull request merged**